packages/bigtable/src/row.js

```diff
@@ -158,25 +123,58 @@ util.inherits(Row, common.GrpcServiceObject);
  * // }
  */
 Row.formatChunks_ = function(chunks) {
-  var families = [];
-  var chunkList = [];
+  var rows = [];
+  var familyName;
+  var qualifierName;
+
+  chunks.reduce(function(row, chunk) {
+    var family;
+    var qualifier;
+
+    row.data = row.data || {};
 
-  chunks.forEach(function(chunk) {
-    if (chunk.resetRow) {
-      chunkList = [];
+    if (chunk.rowKey) {
+      row.key = Mutation.convertFromBytes(chunk.rowKey);
     }
 
-    if (chunk.rowContents) {
-      chunkList.push(chunk.rowContents);
+    if (chunk.familyName) {
+      familyName = chunk.familyName.value;
+    }
+
+    if (familyName) {
+      family = row.data[familyName] = row.data[familyName] || {};
+    }
+
+    if (chunk.qualifier) {
+      qualifierName = Mutation.convertFromBytes(chunk.qualifier.value);
+    }
+
+    if (qualifierName) {
+      qualifier = family[qualifierName] = family[qualifierName] || [];
```

Comment:
It seems problematic that `family` only exists if `familyName` exists. Should these both be in an `if`?

```
if (familyName && qualifierName) {
```